### ISSUE TITLE
Given the parameter `serializer`,  still got exception information `A valid serializer is required`

### ISSUE THREAD

**Issue by ZhangRuiMingZRM** [phalcon/cphalcon#14324]
* Phalcon version: `4.0.0`
* PHP Version: `7.3.7`
* Operating System: ubuntu
* Installation type: Compiling from source
* Server: php web server

**In the document of [Cache Component](https://docs.phalconphp.com/4.0/en/cache#stream)**

```php
use Phalcon\Cache\Adapter\Stream;
use Phalcon\Storage\Serializer\Json;

$jsonSerializer = new Json();

$options = [
    'defaultSerializer' => 'Json',
    'lifetime'          => 7200,
    'serializer'        => $jsonSerializer,    
    'cacheDir'          => '/data/storage/cache',
];

$adapter = new Stream(null, $options);
```

**I got an exception when  running this code.**
```
A valid serializer is required
```

**This is caused by**
https://github.com/phalcon/cphalcon/blob/3d76cdaee5898d9566f59b8996c4ddbe4b29f729/phalcon/Storage/Adapter/AbstractAdapter.zep#L174-L186

I think only if both conditions ( null === this->serializerFactory && null === this->serializer ) are met can throw exception

**This may be a solution**

```zephir
protected function initSerializer() -> void
 {
        string className;

        if unlikely (null === this->serializerFactory && null === this->serializer) {
            throw new Exception("A valid serializer is required");
        }

       if !(typeof this->serializer == "object" && this->serializer instanceof SerializerInterface) {
            let className        = strtolower(this->defaultSerializer),
                this->serializer = this->serializerFactory->newInstance(className);
        }
}
```

**Comment by CameronHall** [comment]
If you're up to it @ZhangRuiMingZRM, feel free to create a PR to address the issue. If not we'll get to it soon :)

**Comment by ruudboon** [comment]
Fixed in 4.0.x branch. thnx @ZhangRuiMingZRM ZhangRuiMingZRM,
